mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -196,6 +225,20 @@ def _init_settings(self):
         self.settings = get_local_settings(settings_read_path, self.name)
         self._initial_settings = deepcopy(self.settings)
 
+    @classmethod
+    def is_compatible(cls):
+        """Is skill supported by current MycroftCore version.
+
+        Skill can read the version info from mycroft.version and determine if
+        it can be loaded. This allows the skill v3 framework to disregard
+        incompatible skill classes in a file and select another one.
+
+        Returns:
+            (bool): True if compatible with the version of Mycroft Core,
+                    else False.
+        """
+        return True
+
```

Comment:
I think this needs some discussion.  We are counting on skill developers to keep this up to date with each version?  I think this is a ticking time bomb.  Some skill won't implement it, or implement it incorrectly, and chaos will ensue.

Reply:
In the general case it should not be overridden by the skill creator, skills should be made compatible with an entire major release and this shouldn't be needed to be implemented.

It is for example for the case if we add a new feature foo in 21.2.7, we want to start using the new feature in the skill. we can then have an `__init__.py`

with

```python
from .skill import SkillWithNewFeature
from .compatibility_skill import CompatibilitySkill
```

compatibility_skill.py

```python
[...]
from mycroft.version import CORE_VERSION_TUPLE

class CompatibilitySkill:
    [...]
    @classmethod
    def is_compatible(cls):
        return CORE_VERSION_TUPLE < (21, 2, 7)
```

And the skill.py:
```python
[...]
from mycroft.version import CORE_VERSION_TUPLE

class SkillWithNewFeature:
```python
from mycroft.version import CORE_VERSION_TUPLE
    [...]
    @classmethod
    def is_compatible(cls):
        return CORE_VERSION_TUPLE >= (21, 2, 7)
```

This has been done earlier with the skill installer skill when the skill-id was added but then the logic was placed in the `create_skill()` function. SInce the suggestion is to remove the need for that method I wanted to suggest a replacement.

The exact same thing can be done with simple imports in the skill __init__

```python
from mycroft.version import CORE_VERSION_TUPLE

if CORE_VERSION_TUPLE < (20, 2, 7):
    from .skill import SkillWithNewFeature
else:
    from .compatibility_skill import CompatibilitySkill
```

My thought was that this was using the classmethod would be more explicit.